core/src/com/projectkorra/projectkorra/waterbending/WaterSpoutWave.java

```diff
@@ -214,8 +215,22 @@ public void progress() {
 				if (isPlant(this.origin.getBlock()) || isSnow(this.origin.getBlock())) {
 					new PlantRegrowth(this.player, this.origin.getBlock());
 					this.origin.getBlock().setType(Material.AIR);
-				} else if (isCauldron(this.origin.getBlock())) {
-					this.origin.getBlock().setType(Material.CAULDRON);
+				} else if (isCauldron(this.origin.getBlock()) || isMud(this.origin.getBlock()) || isSponge(this.origin.getBlock())) {
+					updateSourceBlock(this.origin.getBlock(), block -> {
```

Comment:
What is the point of calling `updateSourceBlock` to a consumer only to run it? I don't understand

Reply:
For some reason the original design was that `WaterSpoutWave` would take the entire liquid content of the cauldron unlike the other abilities which just decrease the level by 1, so this was to provide the freedom to do so. Honestly, I'll just get rid of that entire thing and have a singular `updateSourceBlock` to do what it needs to do because I don't know why `WaterSpoutWave` should be special in that regard.